consensus/types.go

```diff
@@ -1,53 +1,161 @@
 package consensus
 
 import (
+	"sync"
 	"time"
 
 	"github.com/deso-protocol/core/bls"
+	"github.com/deso-protocol/core/collections/bitset"
+	"github.com/holiman/uint256"
 )
 
-// ConsensusEvent is a way for FastHotStuffConsensus to send messages back to
-// the Server. There are three types of events that can be sent:
-//   - Vote: The consensus is ready to vote on a block
+// ConsensusEvent is a way for FastHotStuffConsensus to send messages back to the Server.
+// There are four types of events that can be sent:
+//   - Vote: The consensus is ready to vote on a block at a given block height and view
 //   - Timeout: The consensus has timed out on a view
-//   - BlockProposal: The consensus has a QC for a block and is ready to propose the next
-//     block
+//   - ConstructVoteQC: The consensus has a QC for a block and is ready to construct the
+//     next block at the next block height and the current view
+//   - ConstructTimeoutQC: The consensus has a timeout QC for a view and is ready to construct
+//     an empty block with the timeout QC at the next block height and the current view
```

Comment:
I had a similar comment earlier, maybe we name the events by name? I.e. instead of `Vote` we say `ConsensusEventTypeVote`. This will make the comment more clear to the reader.

Reply:
Ah good catch. I forgot to update these names. Resolved now